core/object/worker_thread_pool.compat.cpp

```diff
@@ -32,12 +32,12 @@
 
 #include "worker_thread_pool.h"
 
-void WorkerThreadPool::wait_for_task_completion_compat_77143(TaskID p_group) {
-	wait_for_group_task_completion(p_group);
+void wait_for_task_completion_compat_77143(WorkerThreadPool *p_this, WorkerThreadPool::TaskID p_group) {
```

Comment:
You seem to use three different approaches for these compat methods in this commit:
- `static` in anonymous namespace
- non `static` in anonymous namespace
- lambda in `_bind_compatibility_methods`

We should probably standardize around one format and try to keep it consistent.

CC @lawnjelly as this might have impact on the SCU build #77170.

Reply:
`non static in anonymous namespace` actually there is no anonymous namespace here that is just me forgetting to put the static on those methods (whoops)

The decisions were use static (ie compilation unit private) methods if possible and fall back to lambdas in the `_bind_compatibility_methods` if access to non public members of the type are necessary (or for some reason that type is entirely defined in the header file and I didn't add a .compat.cpp file)

Reply:
Providing these have unique names after scoping (and things like `wait_for_task_completion_compat_77143()`) looks pretty unique) there's no great problem for the SCU build using static functions confined to a cpp.

While we should generally prefer having funcs scoped by the class ... to be pragmatic, it isn't a hard rule, and I can see the desire to keep the compatibility stuff out of the headers.

What I would point out is that having this stuff in a separate cpp file for a lot of classes, is likely to significantly slow down the regular build (if you e.g. add these to every core class, then rebuilding core will take approx twice as long).

What you could do instead is put the compatibility stuff into `.inc` files (instead of `.cpp`), and include them directly into the regular `.cpp` files. This will still allow the separation, but won't affect the regular build times as much.

Reply:
Re: build times: It's not as bad as double the build time, but it still is pretty bad. For example `editor_interface.cpp` takes 9.5s to compile and `editor_interface.compat.cpp` takes 5.2s and basically all of that is spent parsing headers (most of which is probably useless)

Reply:
Build time is a good consideration yes.

The `.inc` approach sounds like a good tradeoff.

Reply:
`.inc` files sound exactly like what I imagined it should be with my original suggestion, I just didn't know the proper way to describe it :)

Reply:
Moved everything to `*.compat.inc` files and replaced the lambdas, so that all the bound functions in the second variant now also use the same style (tho that meant that I had to make a few methods of `AnimationNode` public that were previously protected for some reasons; those methods were already public in that they are bound in ClassDB and just simple wrappers, so not sure what was going on with them before)